client/src/components/QuestionsList/QuestionsList.component.tsx

```diff
@@ -26,10 +36,15 @@ const QuestionsList = ({ sort, tags, pageSize }) => {
     <Spinner type="page" width="75px" height="200px" />
   ) : (
     <>
-      <PostListComponent posts={data.posts} />
+      <PostListComponent
+        posts={posts?.posts}
```

Comment:
I don't think the use of the name `posts` was helpful so...
```suggestion
        posts={data?.posts}
```